Sources/Bond/AppKit/NSOutlineView+ObservableTree.swift

```diff
@@ -0,0 +1,180 @@
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+#if canImport(AppKit)
+
+import AppKit
+import ReactiveKit
+
+open class OutlineViewBinder<TreeNode: TreeNodeProtocol> {
+    public var insertAnimation: NSOutlineView.AnimationOptions? = [.effectFade, .slideUp]
+    public var deleteAnimation: NSOutlineView.AnimationOptions? = [.effectFade, .slideUp]
+
+    let isItemExpandable: ((TreeNode, TreeNode.Value?, NSOutlineView) -> Bool)?
+    let measureCell: ((TreeNode, TreeNode.Value?, NSOutlineView) -> CGFloat?)?
+    let createCell: ((TreeNode, TreeNode.Value?, NSTableColumn?, NSOutlineView) -> NSView?)?
+
+    public init() {
+        // This initializer allows subclassing without having to declare default initializer in subclass.
+        self.isItemExpandable = nil
+        self.measureCell = nil
+        self.createCell = nil
+    }
+
+    public init(isItemExpandable: ((TreeNode, TreeNode.Value?, NSOutlineView) -> Bool)? = nil, measureCell: ((TreeNode, TreeNode.Value?, NSOutlineView) -> CGFloat?)? = nil, createCell: ((TreeNode, TreeNode.Value?, NSTableColumn?, NSOutlineView) -> NSView?)? = nil) {
+        self.isItemExpandable = isItemExpandable
+        self.measureCell = measureCell
+        self.createCell = createCell
+    }
+
+    open func isExpandable(for item: TreeNode.Value?, outlineView: NSOutlineView, dataSource: TreeNode) -> Bool {
+        return self.isItemExpandable?(dataSource, item, outlineView) ?? false
+    }
+
+    open func height(for item: TreeNode.Value?, outlineView: NSOutlineView, dataSource: TreeNode) -> CGFloat? {
+        return self.measureCell?(dataSource, item, outlineView) ?? outlineView.rowHeight
+    }
+
+    open func cell(for item: TreeNode.Value?, tableColumn: NSTableColumn?, outlineView: NSOutlineView, dataSource: TreeNode) -> NSView? {
+        return self.createCell?(dataSource, item, tableColumn, outlineView) ?? nil
+    }
+
+    open func apply(_ event: ObservableCollectionEvent<TreeNode>, to outlineView: NSOutlineView) {
+        if (self.insertAnimation == nil && self.deleteAnimation == nil) || event.diff.isEmpty {
+            outlineView.reloadData()
+            return
+        }
+
+        let rootNode = event.collection
+        let patch = event.patch
+
+        outlineView.beginUpdates()
+
+        for operation in patch {
+            switch operation {
+            case .insert(let at):
+                let parent = rootNode[at.dropLast()]
+                outlineView.insertItems(at: IndexSet(integer: at.item), inParent: parent, withAnimation: insertAnimation ?? [])
```

Comment:
I don't `at.item` will work here and below. It's probably something like `at.last!` that we need.